enforcer_interface.go

```diff
@@ -107,27 +111,38 @@ type IEnforcer interface {
 	HasPolicy(params ...interface{}) bool
 	HasNamedPolicy(ptype string, params ...interface{}) bool
 	AddPolicy(params ...interface{}) (bool, error)
+	AddPolicyWithContext(ctx context.Context, params ...interface{}) (bool, error)
 	AddPolicies(rules [][]string) (bool, error)
 	AddNamedPolicy(ptype string, params ...interface{}) (bool, error)
+	AddNamedPolicyWithContext(ctx context.Context, ptype string, params ...interface{}) (bool, error)
 	AddNamedPolicies(ptype string, rules [][]string) (bool, error)
 	RemovePolicy(params ...interface{}) (bool, error)
+	RemovePolicyWithContext(ctx context.Context, params ...interface{}) (bool, error)
 	RemovePolicies(rules [][]string) (bool, error)
 	RemoveFilteredPolicy(fieldIndex int, fieldValues ...string) (bool, error)
+	RemoveFilteredPolicyWithContext(ctx context.Context, fieldIndex int, fieldValues ...string) (bool, error)
 	RemoveNamedPolicy(ptype string, params ...interface{}) (bool, error)
+	RemoveNamedPolicyWithContext(ctx context.Context, ptype string, params ...interface{}) (bool, error)
 	RemoveNamedPolicies(ptype string, rules [][]string) (bool, error)
 	RemoveFilteredNamedPolicy(ptype string, fieldIndex int, fieldValues ...string) (bool, error)
+	RemoveFilteredNamedPolicyWithContext(ctx context.Context, ptype string, fieldIndex int, fieldValues ...string) (bool, error)
 	HasGroupingPolicy(params ...interface{}) bool
 	HasNamedGroupingPolicy(ptype string, params ...interface{}) bool
-	AddGroupingPolicy(params ...interface{}) (bool, error)
+	AddGroupingPolicyWithContext(ctx context.Context, params ...interface{}) (bool, error)
```

Comment:
Why is ``AddGroupingPolicy`` removed here?

Reply:
Only by accident, re-adding.